src/snippets/responsive-image.liquid

```diff
@@ -0,0 +1,89 @@
+
+    Usage:
+    In your liquid template file, copy the following line
+    - {% include 'responsive-image' with image: featured_image, img_class: "css-class", wrapper_class: "wrapper-css-class", width: 700, height: 800, mobile_height: 400, include_wrapper: true %}
+{% endcomment %}
+{% assign mobile_breakpoint = '750px' %}
+
+{% comment %} Added incremental number to avoid conflict styling code when the same images are using this snippet {% endcomment %}
+{% capture responsive_image_counter %}{% increment image_counter %}{% endcapture %}
+
+<style>
+  {% if image.aspect_ratio <= 1 %}
+    {% assign max_height = image.height | at_most: height %}
+    {% assign max_width = max_height | times: image.aspect_ratio %}
+  {% else %}
+    {% assign max_width = image.width | at_most: width %}
+    {% assign max_height = max_width | divided_by: image.aspect_ratio %}
+  {% endif %}
+
+  #ProductImage-{{ image.id }}-{{ responsive_image_counter }} {
+    max-width: {{ max_width }}px;
+    max-height: {{ max_height }}px;
+  }
+  #ProductImageWrapper-{{ image.id }}-{{ responsive_image_counter }} {
+    max-width: {{ max_width }}px;
+  }
+
+  {% comment %}
+    Calculate max height and widths again for small screens
+  {% endcomment %}
+
+  {% if image.aspect_ratio <= 1 %}
+    {% assign max_height = image.height | at_most: mobile_height %}
+    {% assign max_width = max_height | times: image.aspect_ratio %}
+  {% else %}
+    {% assign max_width = image.width | at_most: width %}
+    {% assign max_height = max_width | divided_by: image.aspect_ratio %}
+  {% endif %}
+
+  @media screen and (max-width: {{ mobile_breakpoint}} ) {
+    #ProductImage-{{ image.id }}-{{ responsive_image_counter }} {
+      max-width: {{ max_width }}px;
+      max-height: {{ max_height }}px;
+    }
+    #ProductImageWrapper-{{ image.id }}-{{ responsive_image_counter }} {
+      max-width: {{ max_width }}px;
+    }
+  }
+</style>
+
+{% assign img_url = image | img_url: '1x1' | replace: '_1x1.', '_{width}x.' %}
+
+{% if include_wrapper %}
+  <div id="ProductImageWrapper-{{ image.id }}-{{ responsive_image_counter }}" class="responsive-image__container">
+    <div data-image-id="{{ image.id }}" class="responsive-image__wrapper {{ wrapper_class }}" style="padding-top:{{ 1 | divided_by: image.aspect_ratio | times: 100}}%;">
+{% endif %}
+    <img id="ProductImage-{{ image.id }}-{{ responsive_image_counter }}"
+      class="{% if include_wrapper %}responsive-image__image{% endif %} lazyload {{ img_class }}"
+      src="{{ image | img_url: '300x' }}"
+      data-src="{{ img_url }}"
+      data-widths="[360, 540, 720, 900, 1080, 1296, 1512, 1728, 1944, 2048, 4472]"
+      data-aspectratio="{{ image.aspect_ratio }}"
+      data-sizes="auto"
+      alt="{{ image.alt | escape }}"
+      {{ html_attributes }}>
+{% if include_wrapper %}
+    </div>
+  </div>
+{% endif %}
+
+<noscript>
+  <img class="{{ img_class }}" src="{{ image | img_url: '2048x2048' }}" alt="{{ image.alt | escape }}">
```

Comment:
Maybe we should include the wrapper in the fallback here?

Reply:
Currently, it's barebone image. The `<div>` is only needed when we interact with lazysizes. But now that you mention it, I will add the srcset!

Reply:
@t-kelly Actually, I remoed the <noscript> image but the default one is already showing correctly. Here's how it looks like when JS is disabled : 

<img width="1984" alt="noscript_preview" src="https://user-images.githubusercontent.com/658169/38314584-8f530a02-37f4-11e8-8af7-668611cbae56.png">

Reply:
After a small research, I'm putting it back but we need to talk regarding the implementation.